graph/path/earliest.go

```diff
@@ -0,0 +1,100 @@
+package path
+
+import (
+	"gonum.org/v1/gonum/graph"
+	"log"
+	"gonum.org/v1/gonum/graph/temporal"
+)
+
+type Earliest struct {
+	from  graph.Node
+	at    uint64
+	until uint64
+	nodes map[int64]struct{
+		earliest uint64
+		via      []int64
+	}
+}
+
+func (e *Earliest) set(v graph.Node, t uint64, p []int64) {
+	e.nodes[v.ID()] = struct{
+		earliest uint64
+		via      []int64
+	}{
+		t,
+		p,
+	}
+}
+
+func (e *Earliest) From() graph.Node { return e.from }
+
+func (e *Earliest) At() uint64 { return e.at }
+
+func (e *Earliest) Until() uint64 { return e.until }
+
+func (e *Earliest) To(u interface{}) (path []graph.Node, duration uint64) {
+	var uid int64
```

Comment:
So this type switch goes away and the import of "log" (we don't log in libraries).